### ISSUE TITLE
lint: check unnamed returns with similar type

### ISSUE THREAD

**Issue by cristaloleg** [go-critic/go-critic#207]
before:
```go
func f() (float64, float64)
```
after:
```go
func f() (x, y float64)
```
This will make function definition much easier to read & understand.
We might also be interested to warn users about unnamed params/returns in method declaration too.

**Comment by Quasilyte** [comment]
Wow. Good one.

**Comment by Quasilyte** [comment]
Worth referencing https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters.
